src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService_FindDeclaration.cs

```diff
@@ -71,7 +71,15 @@ public bool CanAddTo(SyntaxNode destination, Solution solution, CancellationToke
                 return false;
             }
 
-            // check for generated files if needed.
+            // We can never generate into a document from a source generator, because those are immutable
+            if (document is SourceGeneratedDocument)
+            {
+                return false;
+            }
```

Comment:
why not have SourceGeneratedDocument return true for IsGeneratedCode?

Reply:
We could do that _too_, but it's critical this check happens no matter what the "checkGeneratedCode" flag is set to. I pushed a comment update probably while you were reviewing which might clarify. (Or might not.)

Reply:
ah, that's fair.